### ISSUE TITLE
py2/3 compatibility?

### ISSUE THREAD

**Issue by bdewilde** [dreyco676/pytrends#7]
This package is only Python3-compatible, and its predecessor was only Python2-compatible. Any interest in making this work in both Python versions? I'm happy to hack on it and submit a PR.

I should note that if this is made Py2/3 compatible, a [universal wheel](https://packaging.python.org/en/latest/distributing.html#universal-wheels) could be used if/when registering with PyPi.

**Comment by dreyco676** [comment]
That'd be sweet. I've got a bit more experience since I started porting, let me know if you get stuck or want some help.

**Comment by bdewilde** [comment]
Thanks for adding me in as a contributor! I'll keep the pull requests coming. :+1:

**Comment by dreyco676** [comment]
No problem!

**Comment by bdewilde** [comment]
I've created a new branch — `py23-compatibility` — and am working on this. Rather than messing with try/except imports, I added a dependency on the great [future](http://python-future.org/index.html) package, which conveniently deals with the reorganization of the standard library and builtin object differences between Python 2 and 3. However, I'm getting an error when running `pyGTrends` in Python 2, and it's coming from somewhere within the `urllib` + `cookielib` libraries. Since you figured out how to port that code to Python 3, I'm hoping you'd know how to fix this:

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-3-23fe523a3992> in <module>()
----> 1 connector = pyGTrends.pyGTrends(google_user, google_pass)

/Users/bjdewilde/.pyenv/versions/2.7.9/lib/python2.7/site-packages/pytrends-0.1.1-py2.7.egg/pytrends/pyGTrends.pyc in __init__(self, username, password)
     42         self.url_PrefCookie = "http://www.google.com"
     43         self.header_dictionary = {}
---> 44         self._connect()
     45 
     46     def _connect(self):

/Users/bjdewilde/.pyenv/versions/2.7.9/lib/python2.7/site-packages/pytrends-0.1.1-py2.7.egg/pytrends/pyGTrends.pyc in _connect(self)
     59         galx = re.compile('<input name="GALX"[\s]+type="hidden"[\s]+value="(?P<galx>[a-zA-Z0-9_-]+)">')
     60 
---> 61         resp = self.opener.open(self.url_ServiceLoginBoxAuth).read()
     62         resp = re.sub(r'\s\s+', ' ', bytes.decode(resp))
     63 

/Users/bjdewilde/.pyenv/versions/2.7.9/lib/python2.7/site-packages/future/backports/urllib/request.pyc in open(self, fullurl, data, timeout)
    498         for processor in self.process_response.get(protocol, []):
    499             meth = getattr(processor, meth_name)
--> 500             response = meth(req, response)
    501 
    502         return response

/Users/bjdewilde/.pyenv/versions/2.7.9/lib/python2.7/site-packages/future/backports/urllib/request.pyc in http_response(self, request, response)
   1342 
   1343     def http_response(self, request, response):
-> 1344         self.cookiejar.extract_cookies(response, request)
   1345         return response
   1346 

/Users/bjdewilde/.pyenv/versions/2.7.9/lib/python2.7/cookielib.pyc in extract_cookies(self, response, request)
   1640             self._policy._now = self._now = int(time.time())
   1641 
-> 1642             for cookie in self.make_cookies(response, request):
   1643                 if self._policy.set_ok(cookie, request):
   1644                     _debug(" setting cookie: %s", cookie)

/Users/bjdewilde/.pyenv/versions/2.7.9/lib/python2.7/cookielib.pyc in make_cookies(self, response, request)
   1557         # get cookie-attributes for RFC 2965 and Netscape protocols
   1558         headers = response.info()
-> 1559         rfc2965_hdrs = headers.getheaders("Set-Cookie2")
   1560         ns_hdrs = headers.getheaders("Set-Cookie")
   1561 

AttributeError: 'HTTPMessage' object has no attribute 'getheaders'
```

Any ideas?

**Comment by dreyco676** [comment]
I'll have to dig into this next week. I do remember the biggest change was that in python 2 it handled the response differently (most likely because python 2 is not unicode based). It might be that the response is encoded differently than what its expecting. I can't confirm until I dig in more.

**Comment by bdewilde** [comment]
Thanks for the offer! I think I figured it out, but the solution required dropping `future` in favor of checking the version of Python (2 or 3) and importing libraries accordingly. See commit 613addb7caf13d724253e0ea7fae53fbfccf45e9. I also had to be more careful about bytes vs. unicode for Python 2. And I ended up refactoring this whole part out into its own module... Pull Request to come, but it might be another couple days.
